test/Ocelot.UnitTests/Polly/PollyCircuitBreakingDelegatingHandlerTests.cs

```diff
@@ -131,23 +161,215 @@ public FakeAsyncPolicy(string name, object result)
             Result = result;
         }
 
-        protected override async Task<TResult> ImplementationAsync<TResult>(Func<Context, CancellationToken, Task<TResult>> action,
-            Context context, CancellationToken cancellationToken, bool continueOnCapturedContext)
+        protected override async Task<TResult> ImplementationAsync(Func<Context, CancellationToken, Task<TResult>> action, Context context, CancellationToken cancellationToken,
+            bool continueOnCapturedContext)
         {
             Times++;
             Method = action.Method;
             Target = action.Target;
 
             if (IsLast)
             {
-                TResult r = Result?.GetType() == typeof(TResult)
+                var r = Result?.GetType() == typeof(TResult)
                     ? (TResult)Result
                     : Activator.CreateInstance<TResult>();
                 return r;
             }
 
-            TResult result = await action(context, cancellationToken);
+            var result = await action(context, cancellationToken);
             return result;
         }
+
+        public new IAsyncPolicy WithPolicyKey(string policyKey)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task ExecuteAsync(Func<Task> action)
+        {
+            throw new NotImplementedException();
+        }
```

Comment:
Strange redundant empty methods because of the `IAsyncPolicy` interface implementation.
Could we use `Mock<IAsyncPolicy>` or `Mock<AsyncPolicy<TResult>>` instead of `FakeAsyncPolicy<TResult>`?
Is it possible to use mocks to test?

Reply:
Yeah, ok, that's the original code, let me check that

Reply:
If refactoring requires a lot of effort, then it is better to ignore it.